Mathlib/Util/CountHeartbeats.lean

```diff
@@ -24,6 +24,49 @@ open Lean Elab Command Meta
 namespace Mathlib.CountHeartbeats
 
 
+
+open Tactic
+
+/--
+Run a tactic, optionally restoring the original state, and report just the number of heartbeats.
+-/
+def runTacForHeartbeats (tac : TSyntax `Lean.Parser.Tactic.tacticSeq) (revert : Bool := true) :
+    TacticM Nat := do
+  let start ← IO.getNumHeartbeats
+  let s ← saveState
+  evalTactic tac
+  if revert then restoreState s
+  return (← IO.getNumHeartbeats) - start
+
+/--
+Given a `List Nat`, log an info message with the minimum, maximum, and standard deviation.
+-/
+def logVariation {m} [Monad m] [MonadLog m] [AddMessageContext m] [MonadOptions m]
+    (counts : List Nat) : m Unit := do
+  let counts := counts.map fun i => i / 1000 -- convert to user-facing heartbeats
+  let counts := counts.toArray.qsort (· < ·)
+  let μ := counts.foldl (· + · ) 0 / counts.size
+  -- We jump through some hoops here to get access to `Float.sqrt`, to avoid imports.
+  let var := (Float.sqrt <| UInt64.toFloat <| UInt64.ofNat
+    ((counts.map fun i => (i - μ)^2).foldl (· + · ) 0)).toUInt64.toNat
+  let stddev := var * 100 / μ
+  logInfo s!"Min: {counts[0]!} Max: {counts[counts.size - 1]!} StdDev: {stddev}%"
```

Comment:
```suggestion
def variation (counts : List Nat) : List Nat :=
  let min := (counts.minimum?.getD 0)
  let max := (counts.maximum?.getD 0)
  let μ := counts.foldl (· + · ) 0 / counts.length
  -- We jump through some hoops here to get access to Float.sqrt, to avoid imports.
  let stddev := (Float.sqrt <| UInt64.toFloat <| UInt64.ofNat
    -- `i - μ + (μ - i)` really computes `|i - μ|` with non-truncated subtraction
    (((counts.map fun i => (i - μ + (μ - i))^2).foldl (· + ·) 0) / counts.length)).toUInt64.toNat
  [min, max, stddev]

def logVariation {m} [Monad m] [MonadLog m] [AddMessageContext m] [MonadOptions m]
    (counts : List Nat) : m Unit := do
  if let [min, max, stddev] := variation counts then
  -- convert `[min, max, stddev]` to user-facing heartbeats
  logInfo s!"Min: {min / 1000} Max: {max / 1000} StdDev: {stddev / 10}%"
```
Should `logVariation` be decoupled from the MonadLog?

Also, I think that the current implementation of stddev is not correct:
```lean
run_cmd
  let lis := [0, 10000]
  logVariation lis
  logVariation <| lis ++ lis
/-
Min: 0 Max: 10 StdDev: 100%
Min: 0 Max: 10 StdDev: 140%
-/
```